### PR TITLE
Update Release docs

### PR DESCRIPTION
**Pull request by Warfields**
Added explanation on why release actions are done on main, not on the release tag.

### CODE REVIEW

**Review by hoodmane — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ Once all binaries (5 platforms) have been automatically added to the release (fe
 - navigate to "Actions" github tab
 - pick "Publish to NPM" workflow
 - *keep* main branch
+  - This must be main, not the version tag. The workflow is privileged and requires access to github
```
Comment (garrettgu10) [inline RELEASE.md:25]:
```suggestion
  - This must be main, not the version tag. The workflow requires access to GitHub
```

File: RELEASE.md [hunk 2]
```diff
@@ -22,6 +22,9 @@ Once all binaries (5 platforms) have been automatically added to the release (fe
...
+  repo secrets, so it can only be executed from a protected branch like main, otherwise it can't
+  get the required API keys
```
Comment (garrettgu10) [inline RELEASE.md:27]:
```suggestion
  get the required npm API key.
```

**Comment by fhanau** [comment]
I actually don't think this is correct – at least the `Publish to NPM` action has successful runs from a tag. I think avoiding secrets being misused is intended to already be ensured by only the team being allowed to run these actions manually.

**Comment by fhanau** [comment]
In fact, maybe we **should** recommend to use the release tag in case there are changes on main that add new APIs that aren't part of the actual release yet.

**Review by fhanau — APPROVED** [review]

**Comment by garrettgu10** [comment]
Shoild we also update guidance for workers-types?

**Comment by Warfields** [comment]
I agree with @fhanau.

> Shoild we also update guidance for workers-types?

Yes.

**Pull request merged**